servers/src/main/java/tachyon/worker/block/evictor/LRUEvictor.java

```diff
@@ -152,31 +155,57 @@ protected StorageDirView cascadingEvict(long bytesToBeAvailable, BlockStoreLocat
       return null;
     }
 
-    // 4. cascading eviction: try to free space in next tier to move candidate blocks there, evict
-    // blocks only when it can not be moved to next tiers
+    // 4. cascading eviction: try to allocate space in the next tier to move candidate blocks
+    // there. If allocation fails, the next tier will continue to evict its blocks to free space.
+    // Blocks are only evicted from the last tier or it can not be moved to the next tier.
     candidateDirView = dirCandidates.candidateDir();
     List<Long> candidateBlocks = dirCandidates.candidateBlocks();
-    List<StorageTierView> tierViewsBelow =
-        mManagerView.getTierViewsBelow(candidateDirView.getParentTierView().getTierViewAlias());
-    // find a dir in below tiers to transfer blocks there, from top tier to bottom tier
-    StorageDirView candidateNextDir = null;
-    for (StorageTierView tierView : tierViewsBelow) {
-      candidateNextDir =
-          cascadingEvict(dirCandidates.candidateSize() - candidateDirView.getAvailableBytes(),
```

Comment:
The blocks to evict are not changed. I still get the blocks from `dirCandidates.candidateBlocks()`. The difference is that I move the blocks to next tier one by one so the blocks are likely moved to different dirs instead of a single dir.